web/src/ui_util.ts

```diff
@@ -22,17 +22,27 @@ export function place_caret_at_end(el: HTMLElement): void {
     }
 }
 
-export function replace_emoji_with_text($element: JQuery): void {
-    $element
-        .find(".emoji")
-        .text(function () {
-            if ($(this).is("img")) {
-                return $(this).attr("alt") ?? "";
-            }
-            return $(this).text();
-        })
-        .contents()
-        .unwrap();
+export function replace_emoji_name_with_unicode_hex($element: JQuery): void {
+    $element.find("span.emoji").each(function () {
+        const emoji_class: string | undefined = $(this).attr("class");
+
+        if (emoji_class === undefined) {
+            return; // Skip this iteration if class is undefined
+        }
```

Comment:
These comments are useless; you should be explaining what it **means** that the emoji class is undefined. Anyone reading the code can read what the code does, which is what your comment repeats.

Also comments belong on their own line.

Reply:
I have added comments on their own lines and made them make more sense.